src/main/kotlin/com/strumenta/kolasu/model/Processing.kt

```diff
@@ -33,15 +33,8 @@ fun Node.assignParents() {
 /**
  * Recursively execute "operation" on this node, and all nodes below this node.
  */
-fun Node.processNodes(operation: (Node) -> Unit) {
-    operation(this)
-    containmentProperties.forEach { p ->
-        val v = p.get(this)
-        when (v) {
-            is Node -> v.processNodes(operation)
-            is Collection<*> -> v.forEach { (it as? Node)?.processNodes(operation) }
-        }
-    }
+fun Node.processNodes(operation: (Node) -> Unit, walker: KFunction1<Node, Sequence<Node>> = Node::walk) {
```

Comment:
Should we add a comment to explain what walker does? I assume it cotnrols the order in which we traverse nodes

Reply:
Done